src/components/navigation/Header.svelte

```diff
@@ -19,13 +24,80 @@
 	function toggleNavbar() {
 		isMobileMenuOpen = !isMobileMenuOpen;
 	}
+
+	function checkOverflow() {
+		if (!navContainer) return;
+
+		const navWidth = navContainer.clientWidth;
+		const logoWidth = navContainer.querySelector('.logo-container')?.clientWidth || 0;
+		const themeWidth = 48;
+		const availableWidth = navWidth - logoWidth - themeWidth - 32;
+
+		let linksWidth = 0;
+		if (linksContainer) {
+			linksWidth = linksContainer.scrollWidth;
+			linksWidthCache = linksWidth;
+		} else {
+			linksWidth = linksWidthCache;
+		}
+
+		const newShouldShowMobileMenu = linksWidth > availableWidth;
+
+		if (shouldShowMobile !== newShouldShowMobileMenu) {
+			shouldShowMobile = newShouldShowMobileMenu;
+		}
+	}
+
+	function measureLinksWidth() {
+		if (headerLinks && !linksContainer) {
```

Comment:
Use an early return instead of deep nesting:

```
if (!headerLinks || linksContainer) {
  return;
}
```

Note: this logic ^^ may be incorrect! You'll need to test and validate!

Reply:
Yes, correct. I have updated the function to use an early return to avoid deep nesting and make the code more readable.